kedro/ipython/__init__.py

```diff
@@ -187,6 +192,20 @@ def _find_kedro_project(current_dir: Path) -> Any:  # pragma: no cover
     return None
 
 
+def _guess_run_environment() -> str:  # pragma: no cover
+    """Best effort to guess the IPython/Jupyter environment"""
+    # https://github.com/microsoft/vscode-jupyter/issues/7380
+    if os.environ.get("VSCODE_PID") or os.environ.get("VSCODE_CWD"):
+        return "vscode"
+    elif _is_databricks():
+        return "databricks"
+    elif hasattr(get_ipython(), "kernel"):  # type: ignore[no-untyped-call]
```

Comment:
Again, mypy complains a lot about untype calls. It would be great if we can set this to `kedro` related libraries only, because we have no control of 3-party libraries. If they don't use type hint we cannot do anything here.

Reply:
Yeah that sounds reasonable